text/1000-migrations.rst

```diff
@@ -174,6 +174,20 @@ name is as follows::
 First revision must have a parent version of `initial`.
 
 
+Migration operation classification
+==================================
+
+The migrations system classifies the migrations operations into two categories:
+safe and unsafe, based on whether the operation is automatically reversible
+without losing any prior data.  For example, all ``CREATE`` operations are
+considered safe by definition, but also alterations to schema that doesn't
```

Comment:
The "reversible" word reads ambiguous. I assume it's safe within one migration? Because  if during a migration a user adds commands to copy data to the newly created type it's no longer safe to reverse it.

Reply:
The word "prior" is key.  The use case here is the ability to rollback a failed deploy, not the ability to rollback a random migration 2 years after the fact.  Otherwise all, but the most trivial changes to the schema would be "safe".

Reply:
Maybe more accent/clarification on this, like "without loosing data that was already committed before applying migration".

Reply:
OK, will clarify.